### ISSUE TITLE
Bugs inside the DTag transferring process

### ISSUE THREAD

**Issue by RiccardoM** [desmos-labs/desmos#296]
## Bug description
During the completion of the Desmos Primer Phase 6 challenge that required them to perform a DTag transferring, the Discord user MikeSnowman75 told me they ran into a strange behavior that did not allow him to exchange the DTag properly. 

## Steps to reproduce
Reporting his words, here is what he did with an additional analyses from me. 

He tried to transfer the dtag first in these 2 transactions:
1. [First](http://lcd.morpheus.desmos.network:1317/txs/77D58E64007C98A8B6B0A1AF5D0B9E7E03CD140B7094BA96057C497E5E15EC6F)
2. [Second](http://lcd.morpheus.desmos.network:1317/txs/DD2A9293CC66807BFC190553C936E944C37BE910107D7AEC6792B3B81715E8D4)

As we can already see, there are two problems here: 

1. The first transaction has an empty `dtag_to_trade`.  
   This might say that the sending profile did not exist, or it didn't have the DTag set. This is a bug I've already reported inside #292. 

2. The second message fails with error `profile of desmos1k7q92w7ku72ycutz6xqrx67v92yycff3emz88s doesn't exist`.  
   This should not have happened. If a user wants a DTag of another user without having created a profile first, then a new profile for the receiving user should be created and the DTag transferring should continue properly.

After this, he tried with other two transactions: 

1. [First](http://lcd.morpheus.desmos.network:1317/txs/3B68A9CC4F72FE51C68F48A6E4F487EDAE1BA8C6D6AE150B071E1B7E9FE34584)

2. [Second](http://lcd.morpheus.desmos.network:1317/txs/6D819AB4F4B30E5D4C2F266065D0C1770AB5354315DC6F0A62E788B98372ED59)

From them, we can see that: 

1. The first transaction was successful. 
2. The second transaction failed with error `the owner's DTag is different from the one to be exchanged`.  
   Apparently, something here messed up because this one should not have failed here. 

Also, they said that they tried performing [another transaction](http://lcd.morpheus.desmos.network:1317/txs/E3CA26225BD97E2886FCF5C405A3C5091C43166BDE6AA9ED66C7A2F1775EE075). As you can see, this failed with error: 

```
the transfer request from desmos1rpurg200edtv6rgvfjuf7j6s8mghnsljkzvurg to desmos1k7q92w7ku72ycutz6xqrx67v92yycff3emz88s has already been made
```

To be honest, I don't think this error should ever verify. If an Alice -> Bob transfer was made successfully, then once it concluded it should be removed allowing Alice to send Bob another DTag if she wants. I've reported this bug inside #295.

## Expected behavior
The first transferring should have completed properly, as well as the second one.

**Comment by bragaz** [comment]
I think I've understood why this [txs](http://lcd.morpheus.desmos.network:1317/txs/6D819AB4F4B30E5D4C2F266065D0C1770AB5354315DC6F0A62E788B98372ED59) error occurred.
Since in `v0.12.3` we have the empty DTag transfer bug, when we make a request to an address without a profile, the system saves the request with an empty dTag into the store. Later, that address create a profile with a dTag which will be different from the empty one and if he accept the transfer request the error will show up.

**Comment by RiccardoM** [comment]
@bragaz So, this should be fixed with #303 as well?

**Comment by bragaz** [comment]
@RiccardoM I will add some more integration tests to a new PR that should solve this and #295
